controller/src/beerocks/master/son_management.cpp

```diff
@@ -30,6 +30,34 @@ using namespace beerocks;
 using namespace net;
 using namespace son;
 
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////// Local Module Functions ///////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+/**
+ * @brief Set the channel pool to contain all supported channels.
+ * @param channel_pool_set:  Set containing the current channel pool.
+ * @param channel_pool_size: Size of the given set.
+ * @param database:          Reference to the DB.
+ * @param radio_mac:         MAC address of radio.
+ */
+static void set_pool_to_all_supported_channels(std::unordered_set<uint8_t> &channel_pool_set,
+                                               uint8_t &channel_pool_size, db &database,
+                                               const sMacAddr &radio_mac)
+{
+    LOG(DEBUG) << "Setting channel pool to all channels";
+    channel_pool_set.clear();
+    auto all_channels = database.get_hostap_supported_channels(tlvf::mac_to_string(radio_mac));
```

Comment:
`get_hostap_supported_channels` can fail and return an empty string `std::vector<beerocks::message::sWifiChannel>()`
please add a check here.

this means the function should return `bool` and not `void`

Reply:
Even if `get_hostap_supported_channels` fails and we get an empty `std::vector<beerocks::message::sWifiChannel>()` it won't matter since later on in the DCS task we validate the pool with
```
if (curr_channel_pool.empty()) {
        LOG(ERROR) << "empty channel pool is not supported. please set channel pool for mac="
                   << m_radio_mac;
        return beerocks::eChannelScanStatusCode::INVALID_PARAMS;
}
```
So even if an empty pool is set, the scan will fail with an `INVALID_PARAMS` in `m_last_scan_error_code`

Reply:
You're right,
Updated